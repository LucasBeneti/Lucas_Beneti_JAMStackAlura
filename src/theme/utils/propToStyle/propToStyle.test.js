import propToStyle from './index';

describe('propToStyle()', () => {
  describe('when receives simple argument', () => {
    it("and it's a string", () => {
      const propToStyleRes = propToStyle('display');
      const componentProps = { display: 'flex' };
      const styleResult = propToStyleRes(componentProps);
      expect(styleResult).toEqual({ display: 'flex' });
    });

    it("and it's a number", () => {
      const propToStyleRes = propToStyle('flex');
      const componentProps = { flex: 1 };
      const styleResult = propToStyleRes(componentProps);
      expect(styleResult).toEqual({ flex: 1 });
    });
  });

  describe('when receives an object of breakpoints', () => {
    it('renders only one breakpoint resolution', () => {
      const propToStyleRes = propToStyle('display');
      const componentProps = { display: { xs: 'flex' } };
      const styleResult = propToStyleRes(componentProps);
      expect(styleResult).toMatchSnapshot();
    });

    it('renders more breakpoints resolutions', () => {
      const propToStyleRes = propToStyle('display');
      const componentProps = { display: { xs: 'flex', md: 'grid' } };
      const styleResult = propToStyleRes(componentProps);
      expect(styleResult).toMatchSnapshot();
    });
  });
});
