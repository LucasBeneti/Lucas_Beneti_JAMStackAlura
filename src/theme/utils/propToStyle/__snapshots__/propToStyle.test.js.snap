// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`propToStyle() when receives an object of breakpoints renders more breakpoints resolutions 1`] = `
Array [
  "",
  Array [
    "@media screen and (min-width:",
    "768",
    "px){",
    "display: grid;",
    "}",
  ],
]
`;

exports[`propToStyle() when receives an object of breakpoints renders only one breakpoint resolution 1`] = `
Array [
  "",
]
`;
